csrc/ops/utils.cpp

```diff
@@ -174,111 +174,111 @@ IterType promoteIterType(IterType type1, IterType type2) {
+      extent_is_from_symbolic = false;
+    }
+    extent_val = promoteSize(extent_val, id->extent());
+    if (iter_type.has_value()) {
+      iter_type = promoteIterType(iter_type.value(), id->getIterType());
+    } else {
+      iter_type = id->getIterType();
+    }
 
-      auto start_offset = dom[i]->start();
-      auto stop_offset = dom[i]->stopOffset();
+    auto id_start_offset = id->start();
+    auto id_stop_offset = id->stopOffset();
       // Currently, start is always constant
       NVF_ERROR(
-          start_offset->isConstInt(),
+          id_start_offset->isConstInt(),
           "Invalid IterDomain start: ",
-          start_offset);
+          id_start_offset);
       NVF_ERROR(
-          stop_offset->isConstInt(),
+          id_stop_offset->isConstInt(),
           "Invalid IterDomain stop offset: ",
-          stop_offset);
-      start_offsets[i] =
-          std::max(start_offsets[i], start_offset->evaluate().as<int64_t>());
-      stop_offsets[i] =
-          std::max(stop_offsets[i], stop_offset->evaluate().as<int64_t>());
-    }
+          id_stop_offset);
+      start_offset =
+          std::max(start_offset, id_start_offset->evaluate().as<int64_t>());
+      stop_offset =
+          std::max(stop_offset, id_stop_offset->evaluate().as<int64_t>());
   }
-  for (const auto dim_i : c10::irange(out_domain.size())) {
-    if (extent_vals[dim_i] != nullptr) {
+
+  IterDomain* out_domain = nullptr;
+  if (extent_val != nullptr) {
       NVF_ERROR(
-          iter_types[dim_i].has_value(),
+          iter_type.has_value(),
           "Could not deduce iter type for new tensor view.");
-      out_domain[dim_i] =
+      out_domain =
           IterDomainBuilder(
-              IrBuilder::create<Val>(start_offsets[dim_i], DataType::Index),
-              extent_vals[dim_i])
+              IrBuilder::create<Val>(start_offset, DataType::Index),
+              extent_val)
               .stop_offset(
-                  IrBuilder::create<Val>(stop_offsets[dim_i], DataType::Index))
-              .iter_type(iter_types[dim_i].value())
+                  IrBuilder::create<Val>(stop_offset, DataType::Index))
+              .iter_type(iter_type.value())
               .build();
     } else {
-      out_domain[dim_i] = IterDomainBuilder(
+      out_domain = IterDomainBuilder(
                               FusionGuard::getCurFusion()->zeroVal(),
                               FusionGuard::getCurFusion()->oneVal())
-                              .expanded_extent(expanded_extent_vals[dim_i])
+                              .expanded_extent(expanded_extent_val)
                               .iter_type(IterType::Broadcast)
                               .build();
     }
+  return out_domain;
+}
+
+// Adding these pragmas since gcc-12.2.1
+// incorrectly reports a warning with the use of evaluate
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfree-nonheap-object"
+#endif
```

Comment:
CC @protonu. Please check that this GCC check is in the right place. Specifically, should it instead be at `outIterDomain()`? I checked the original PR https://github.com/NVIDIA/Fuser/issues/643 but I didn't find any mention of this code so I'm not sure what the original problem was.

Reply:
Based on the comment about issues with using `evaluate`, it should be around the new `outIterDomain` function.